### PR TITLE
1875 Response requisition that have the status 'NEW' sync back to mSupply status 'cn'

### LINKED ISSUE #1875: Response requisition that have the status 'NEW' sync back to mSupply status 'cn'
## What went wrong? 😲
 
I think that NEW response requisition should sync back a status of 'sg' instead of 'cn'. A reason for this is that the filter of requisitions that are still pending approval are using "sg" as a one of it's parameter to filter still unapproved requisitions. 

The main reason though would be to have a certain consistency with other requisitions done in mSupply where the status will be 'sg' until a customer invoice would be created for that requisition which from that point would be of status 'cn'. 

omSupply
image

mSupply central server
image
 
## Expected behaviour 🤔
Would it be possible for NEW Requsition in omSupply to send sync status 'sg' until a outbound shipment has been created for that requisition?

## How to Reproduce 🔨

Steps to reproduce the behaviour:
Setup a sync setup
1. Create an internal order to a supplier that is a store using omSupply. Sync. 
2. On the supplier store, sync to receive the order so that a response requisition is created from it.
3. Check that the Response requisition has a status of NEW. Sync. Log to that supplier store on the central server. Go to Customer > Requisitions  
4. See error. These requisitions displays status 'cn'

## Your environment 🌱

 - Version: 
 - Platform:
   - [x] android (tablet)
   - [ ] browser (extra points if you tell us which one)
   - [x] desktop (windows)
   - [ ] desktop (macOS)
   - [ ] server (windows)

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #1875

# 👩🏻‍💻 What does this PR do? 
Change new response requisition status to `sg` when syncing to central and fix tests. 

# 🧪 How has/should this change been tested? 
Send an internal order to another store. Check central to see that requisition status is `sg`.

### CODE REVIEW

**Review by mark-prins — COMMENTED** [review]
I think that's good - will wait to see if someone more familiar with sync wants to have a look before I blindly approve!

**Comment by andreievg** [comment]
Hi, i was going to interject before this PR is made, and was going to suggest a reason why we shouldn't address the issue for now, but realised that explaining all of the detailed reason would be more work then just doing the change.

There is one more change to make this complete please, if requisition has shipment we should send `cn` status.

**Comment by roxy-dao** [comment]
> cf1d074

File: server/service/src/sync/translations/requisition.rs [hunk 1]
```diff
@@ -319,17 +320,24 @@ impl SyncTranslation for RequisitionTranslation {
...
             serial_number: requisition_number,
             name_ID: name_id,
             store_ID: store_id.clone(),
             r#type: to_legacy_type(&r#type),
-            status: to_legacy_status(&r#type, &status).ok_or(anyhow::Error::msg(format!(
-                "Unexpected row requisition status {:?} (type: {:?}), row id:{}",
-                status, r#type, changelog.record_id
-            )))?,
+            status: to_legacy_status(&r#type, &status, has_outbound_shipment).ok_or(
+                anyhow::Error::msg(format!(
```
Comment (andreievg) [inline server/service/src/sync/translations/requisition.rs:336]: fyi, `anyhow::anyhow!(` is the same as` anyhow::Error::msg(format!(`

File: server/service/src/sync/translations/requisition.rs [hunk 2]
```diff
@@ -468,7 +477,8 @@ fn to_legacy_status(
             _ => return None,
         },
         RequisitionRowType::Response => match status {
-            RequisitionRowStatus::New => LegacyRequisitionStatus::Cn,
+            RequisitionRowStatus::New if has_outbound_shipment => LegacyRequisitionStatus::Cn,
```
Comment (andreievg) [inline server/service/src/sync/translations/requisition.rs:480]: nice

**Review by andreievg — APPROVED** [review]
thanks, looks great

**Pull request merged**